### PR TITLE
EOS-18717:[ Mini-Provisioner :S3: cleanup phase] Validate reset phase is done or not.

### PR DESCRIPTION
**Pull request by sachitanands**
this check is added in Cleanup phase as pre-validation of reset phase is done or not.

Signed-off-by: Sachitanand Shelake <sachitanand.shelake@seagate.com>

### CODE REVIEW

File: scripts/provisioning/cleanupcmd.py [hunk 1]
```diff
@@ -149,10 +149,30 @@ def cleanup_haproxy_configurations(self):
...
-    sys.stdout.write(f"Processing {self.name} detect_if_reset_done, TODO: implement it\n")
-    # if reset_not_done
-        # raise S3PROVError("reset needs to be performed before cleanup can be processed\n")
+    """Validate if reset phase has done or not, throw exception."""
+    sys.stdout.write(f"Processing {self.name} detect_if_reset_done\n")
+    # Validate log file cleanup.
+    log_files = ['/var/log/seagate/auth/server/app.log', '/var/log/seagate/s3/s3server-*/s3server.INFO']
+    for fpath in log_files:
+      if os.path.exists(fpath):
+        raise S3PROVError("Stale log files found in system!!!! hence reset needs to be performed before cleanup can be processed\n")
```
Comment (nileshgovande) [inline scripts/provisioning/cleanupcmd.py:158]: Make sure exceptions are moved up properly.

Comment (sachitanands) [reply scripts/provisioning/cleanupcmd.py:158]: done

**Review by nileshgovande — APPROVED** [review]

**Review by nileshgovande — APPROVED** [review]

**Review by pratyush-seagate — APPROVED** [review]
LGTM, +1

**Review by vimalraghwani20 — APPROVED** [review]

File: scripts/provisioning/cleanupcmd.py [hunk 2]
```diff
@@ -149,10 +149,30 @@ def cleanup_haproxy_configurations(self):
...
+    # Validate ldap entry cleanup.
+    self.validate_ldap_account_cleanup()
+    sys.stdout.write(f"Processing {self.name} detect_if_reset_done completed successfully..\n")
+
+  def validate_ldap_account_cleanup(self):
+    """Validate ldap data is cleaned."""
+    account_count=0
+    try :
+      sys.stdout.write("INFO:Validating ldap account entries\n")
+      ldap_action_obj = LdapAccountAction(self.ldap_user, self.ldap_passwd)
```
Comment (nileshgovande) [inline scripts/provisioning/cleanupcmd.py:168]: Start LDAP service if not running. 
Move service start and stop code in setupcmd.py
You may have to edit https://github.com/Seagate/cortx-s3server/pull/768

**Pull request merged**